QModManager/Messages.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+
+namespace QModManager
+{
+    public static partial class QModAPI
+    {
+        [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+        public class MessageReceiverAttribute : Attribute
+        {
+            public QMod Sender;
+
+            public MessageReceiverAttribute(QMod sender)
+            {
+                Sender = sender ?? throw new ArgumentNullException("The provided mod is null!", nameof(sender));
+            }
+            public MessageReceiverAttribute(Assembly senderAssembly)
+            {
+                Sender = GetMod(senderAssembly, true, true) ?? throw new ArgumentException("The provided assembly is not a mod assembly!", nameof(senderAssembly));
+            }
+            public MessageReceiverAttribute(string senderID)
+            {
+                // Should we not want to throw an exception in case the mod is not present?
+                Sender = GetMod(senderID, true, true) ?? throw new ArgumentException($"No mod matching the provided ID \"{senderID}\" was found!", nameof(senderID));
+            }
+        }
+
+        [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+        public class GlobalMessageRecieverAttribute : Attribute
+        {
+            public GlobalMessageRecieverAttribute() { }
+        }
+
+        public static void SendMessage(QMod mod, string message, params object[] data)
```

Comment:
This is going to need an example mod to showcase how to use this function,  
both on part of the sender and especially on the part of the receiver.

Reply:
Maybe something for the wiki...?